### ISSUE TITLE
Heuristic audit: should-fix accessibility issues

### ISSUE THREAD

**Issue by yaelberger-commits** [cds-snc/notification-planning#2235]
### Purpose
Track and resolve lower-severity accessibility issues identified through heuristic reviews or formal audits.

### This feature includes
- Grouped should-fix findings from accessibility reviews
- Improvements that enhance accessibility but are not blocking

### This feature does not include
- Critical WCAG violations
- Non-audit-driven work

### Related issues
- #2235 Should-fix issues from Heuristic UX Review

[Presentation](https://docs.google.com/presentation/d/1ZiFHnM-sQYFrlxVvO_TMv06RY8XTjREo7_PJ39wKTjg/edit?slide=id.g3544ca3fd71_0_0#slide=id.g3544ca3fd71_0_0)
[Issues log Spreadsheet](https://docs.google.com/spreadsheets/d/1H0Iw4xiJQhhkYnBKm8SKbvNDeKBkGCZK7tebhguA-CQ/edit?gid=469696501#gid=469696501)
[Miro of triaged issues ](https://miro.com/app/board/uXjVJTuKbXM=/)
